db/constraints/base.py

```diff
@@ -0,0 +1,72 @@
+from abc import ABC, abstractmethod
+
+from db.columns.operations.select import get_columns_name_from_attnums
+from db.tables.operations.select import reflect_table_from_oid
+
+
+class Constraint(ABC):
+
+    @abstractmethod
+    def add_constraint(self, migration_operation, engine, connection_to_use):
+        pass
+
+    @abstractmethod
+    def constraint_type(self):
+        pass
+
+
+class ForeignKeyConstraint(Constraint):
+    def constraint_type(self):
+        return "foreignkey"
+
+    def __init__(
+            self, name,
+            table_oid,
+            columns_attnum,
+            referent_table_oid,
+            referent_columns_attnum,
+            options
+    ):
+        self.name = name
+        self.table_oid = table_oid
+        self.columns_attnum = columns_attnum
+        self.referent_table_oid = referent_table_oid
+        self.referent_columns = referent_columns_attnum
+        self.options = options
+
+    def add_constraint(self, migration_operation, engine, connection_to_use):
```

Comment:
It's strange to me that you're passing the `Operations` object (`migration_operation`) here. Why not just create it from the engine you have in hand? Passing all of these separately is a bit wonky, since if they're out-of-sync (i.e., if the engine isn't the one used for the `MigrationContext` behind the `Operations` class, weird things could happen (what if the engine is reflecting from some other DB, for example?).